python/pyspark/sql/functions.py

```diff
@@ -2208,7 +2208,8 @@ def pandas_udf(f=None, returnType=None, functionType=None):
     1. SCALAR
 
        A scalar UDF defines a transformation: One or more `pandas.Series` -> A `pandas.Series`.
-       The returnType should be a primitive data type, e.g., :class:`DoubleType`.
+       The returnType should be a primitive data type, e.g., :class:`DoubleType` or
+       arrays of a primitive data type (e.g. :class:`ArrayType`).
```

Comment:
It could now be more than just primitive types, I believe all of `pyspark.type.DataType`s except for MapType, StructType, BinaryType and nested Arrays (that needs to be checked).

Reply:
Should :class:`ArrayType` be :class:`ArrayType(DoubleType)`?

Reply:
Checked nested arrays do not currently work. I'll an explicit test to check that this fails, and when the test starts passing we can update the documentation.